templates/repo/diff/box.tmpl

```diff
@@ -166,6 +166,9 @@
 										<a class="ui basic tiny button" rel="nofollow" href="{{$.BeforeSourcePath}}/{{PathEscapeSegments .Name}}">{{ctx.Locale.Tr "repo.diff.view_file"}}</a>
 									{{else}}
 										<a class="ui basic tiny button" rel="nofollow" href="{{$.SourcePath}}/{{PathEscapeSegments .Name}}">{{ctx.Locale.Tr "repo.diff.view_file"}}</a>
+										{{if and $.Repository.CanEnableEditor $.CanEditFile (not $file.IsLFSFile) (not $file.IsBin)}}
+											<a class="ui basic tiny button" rel="nofollow" href="{{$.HeadRepoLink}}/_edit/{{PathEscapeSegments $.HeadBranchName}}/{{PathEscapeSegments $file.Name}}?hide_create_pr=true&hide_create_new_branch=true&redirect_to={{print $.BackToLink "#diff-" $file.NameHash | QueryEscape}}">{{ctx.Locale.Tr "repo.editor.edit_this_file"}}</a>
```

Comment:
I think the `hide_create_pr=true&hide_create_new_branch=` could all be removed.

The meaning of them is: "the editor is for a special purpose and shouldn't have other operations, it should for editing only".

The meaning of "redirect_to" is: "the editor is for a special purpose and after edit it should redirect back".

So I think at the moment, checking "IsEditingFileOnly (has RedirectTo)" also means "HideXxx=true" (no other operation)

Reply:
But as an editor, more parameters will be more flexible than before. Maybe other places can reuse them in future PRs. i.e. Edit file in the commit files view, then it can create new branch or create PR.

Reply:
Then it should have a better "parameters" system at that time, but not keeping adding "hide_xxx"

When there is a real requirement, the design should be like this: `?editor_mode=editing-only`, but not by a lot of `hide_xxx`.

Reply:
I still think using only "redirect_to" is clear enough for hiding the "create PR/branch" operations. 

Because when it should "redirect back", then the editor page shouldn't create PR/branch.

In this case, less is more, we can leave a comment there, and that why I suggest to use template variable `IsEditingFileOnly = (redirectTo == "")`

I do not think it's worth to make the parameters too complex than it should be.

Reply:
4bd286e7aec79589d81ae0d97d21c1e515fab2ff